pkg/config/setup/process_fallback.go

```diff
@@ -0,0 +1,17 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build !linux
+
+package setup
+
+import (
+	pkgconfigmodel "github.com/DataDog/datadog-agent/pkg/config/model"
+)
+
+// overrideRunInCoreAgentConfig sets the process_config.run_in_core_agent.enabled to false in unsupported environments.
+func overrideRunInCoreAgentConfig(config pkgconfigmodel.Config) {
+	config.Set("process_config.run_in_core_agent.enabled", false, pkgconfigmodel.SourceAgentRuntime)
+}
```

Comment:
What about using GOOS intstead of have 2 implementation ?

```suggestion
func overrideRunInCoreAgentConfig(config pkgconfigmodel.Config) {
    if runtime.GOOS == "windows" {
    	config.Set("process_config.run_in_core_agent.enabled", false, pkgconfigmodel.SourceAgentRuntime)
    }
}
```

(I'm guessing we don't want to set this as `true` on darwin ?)

Reply:
Thanks for the suggestion. I ended up doing `GOOS != "linux`. Let me know what you think!